pulsar-client-reactive-api/src/main/java/org/apache/pulsar/reactive/client/api/ReactiveMessageSenderBuilder.java

```diff
@@ -28,156 +28,555 @@
 import org.apache.pulsar.client.api.MessageRoutingMode;
 import org.apache.pulsar.client.api.ProducerAccessMode;
 import org.apache.pulsar.client.api.ProducerCryptoFailureAction;
+import org.reactivestreams.Publisher;
 
+/**
+ * Builder interface for {@link ReactiveMessageSender}.
+ *
+ * @param <T> the message payload type
+ * @author Lari Hotari
+ * @author Christophe Bornet
+ */
 public interface ReactiveMessageSenderBuilder<T> {
 
+	/**
+	 * Sets the cache to use for the sender.
```

Comment:
The ReactiveMessageSenderCache will need a better explanation, but we could do that in a separate PR.

Reply:
Yes I'm doing it interface by interface 🙂